PE-296 - add nodejs multibranch pipeline + small cleanup

lub0v-parsable opened the pull request:
Added Node.js multibranch pipeline. For now, it only tests the app and publishes test & coverage reports.

Renamed catalog to be more generic, I suppose we'll use this repo for almost every template we have. If we see that it's too big we'll split it into different repositories.

Renamed performanceTemplate to just performance for simplicity

kavita-narang commented:
I think the performance template got disconnected from the template job after the changes-
![image](https://user-images.githubusercontent.com/43971253/80768767-27126800-8b00-11ea-85fa-7a371362c67b.png)

lub0v-parsable commented:
> I think the performance template got disconnected from the template job after the changes-
> ![image](https://user-images.githubusercontent.com/43971253/80768767-27126800-8b00-11ea-85fa-7a371362c67b.png)

Yes, I know :(
I will need to re-create it after the merge. It is because I changed the name of the catalog. I've added a warning now to not do it later.

kavita-narang commented:
Can we recreate the template and job now to check if the new updates work?
